values.yaml

```diff
@@ -41,6 +41,25 @@ serviceAccount:
 ## StackStorm shared variables
 ##
 st2:
+  # Username, used to login to StackStorm system
+  username: st2admin
+  # Password, used to login to StackStorm system (auto-generated by default)
+  # The password (set here or auto-generated) is preserved across upgrades.
+  #password: Ch@ngeMe
+  # To force the password to reset (using st2.password, or a newly generated one),
+  # you can use `helm upgrade --set st2.resetPassword=true`.
+  resetPassword: false
```

Comment:
I think the additional `resetPassword` functionality is a bit excessive and would be confusing.

The expected behavior is to always update the password or SSH key once the user changes it via the Helm values.

Reply:
Check the https://github.com/helm/helm/issues/3053#issuecomment-782837107 which tries to query the "previous" pass value instead of relying on `.Release.IsUpgrade`, it should help.

I think the expected and predictable behavior:
- No password/SSH key provided in Helm values = autogenerate it once, keep the state.
- If user changes Helm values unsetting the SSH key/password = just keep the previous state value, do nothing.
- If password/SSH key set in Helm values = always overwrite it with the new value.

WDYT? What are the other corner cases?

Reply:
It sounds like you're thinking of something slightly different than the linked comment. Something like this, right?

```
{{- $previous := lookup "v1" "Secret" .Release.Namespace "foobar-secret" }}
data:
{{- if and $previous (not .Values.st2.password) }}
  password: {{ $previous.data.password }}
{{ else }}
  password: {{ default (randAlphaNum 12) .Values.st2.password | b64enc | quote }}
{{ end }}
```

And then similar logic for the ssh key.

Do we need to support rotating the auto-generated secrets? Or do we say that we only generate once, and any secret rotation must be managed via helm values?

Reply:
Yeah, I think rotation is the user's responsibility. It's already really good enough to have them auto-generated :)

Reply:
K. I removed `resetPassword` and updated the logic around password and ssh_key secrets.